src/library/basesqltablemodel.h

```diff
@@ -139,6 +139,23 @@ class BaseSqlTableModel : public QAbstractTableModel, public TrackModel {
         }
         int m_column;
         Qt::SortOrder m_order;
+        
+        friend QTextStream& operator >> (QTextStream &in, SortColumn &sc) {
+            int ord;
+            in >> sc.m_column >> ord;
+            
+            if (ord > 0) sc.m_order = Qt::AscendingOrder;
+            else sc.m_order = Qt::DescendingOrder;
+            return in;
+        }
+        
+        friend QTextStream & operator << (QTextStream &out, SortColumn &sc) {
+            out << sc.m_column << " ";
```

Comment:
This is because QTextStream skips spaces when reading but you need to set spaces when writing (like it happens normally in cin and cout)